unit_test/CMakeLists.txt

```diff
@@ -34,6 +37,47 @@ TARGET_INCLUDE_DIRECTORIES(kokkoskernels_gtest PUBLIC $<BUILD_INTERFACE:${GTEST_
 #Gtest minimally requires C++11
 TARGET_COMPILE_FEATURES(kokkoskernels_gtest PUBLIC cxx_std_11)
 
+
+
+#
+#
+#  Making some CMake experiments to design a more automated
+#  unit-test framework that generates automatically cpp files
+#  needed by each backend to generate the unit-test executables.
+#
+#
+
+set(MY_COMPS Common)
+set(MY_BACKENDS Serial OpenMP)
+
+foreach(COMP ${MY_COMPS})
+  string(TOLOWER ${COMP} COMP_LOWER)
+  file(GLOB COMP_CPP ${COMP_LOWER}/*.hpp)
+  message(STATUS "List of Test_${COMP}_.cpp files:")
+  foreach(BACKEND ${MY_BACKENDS})
+    string(TOLOWER ${BACKEND} BACKEND_LOWER)
+    string(TOUPPER ${BACKEND} BACKEND_UPPER)
+    if(KOKKOS_ENABLE_${BACKEND_UPPER})
+      foreach(COMP_FILE ${COMP_CPP})
+        get_filename_component(COMP_NAME ${COMP_FILE} NAME_WE)
+        string(REPLACE "Test_${COMP}_" "" COMP_TEST ${COMP_NAME})
```

Comment:
It looks like you only really use `COMP_TEST` in the float128 check. You could probably avoid the replace and just work with `COMP_NAME`.

Reply:
Actually I would rather try to skip `COMP_NAME` than `COMP_TEST`, `COMP_TEST` makes it really easy to write the path, filename and file content on line 69. If I use `COMP_NAME` instead I will have to do more complex string manipulation to generate the correct path, filename and content. It seems more prone to error to me, what do you think?